src/app/service/game-control.service.ts

```diff
@@ -0,0 +1,30 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+
+export type Players = {
+  [key: string]: number;
+};
+
+@Injectable({
+  providedIn: 'root',
+})
+export class GameControlService {
+  private board: BehaviorSubject<number[]> = new BehaviorSubject<number[]>(
+    new Array(9)
+  );
+  private moves: BehaviorSubject<number[]> = new BehaviorSubject<number[]>([]);
```

Comment:
Why 2 streams?

Reply:
one for board, second for player moves

Reply:
Isn't it enough to have one stream?  {3:X}----{1:O}----{5:X}---->